pywwt/annotation.py

```diff
@@ -252,6 +255,100 @@ def _on_trait_change(self, changed):
         super(Line, self)._on_trait_change(changed)
 
 
+class FieldOfView():
+    """
+    A collection of polygon annotations. Takes the name of a pre-loaded 
+    telescope and displays its field of view.
+    """
+    # links for kepler. guide: https://keplerscience.arc.nasa.gov/k2-fields.html
+    # footprint: https://raw.githubusercontent.com/KeplerGO/K2FootprintFiles/master/json/k2-footprint.json
+
+    # more efficient method than CircleCollection of changing trait values?
+    def __init__(self, parent, telescope, center, rot, **kwargs):
+        self.parent = parent
+        self.available = self.parent.instruments.available
+        self.dim = self.parent.instruments.dim
+        
+        # list of IDs of annotations created in this FieldofView instance
+        self.active = []
+        self._gen_fov(telescope, center, rot, **kwargs)
+
+    def _gen_fov(self, telescope, center, rot, **kwargs):
+        #telescope = telescope.lower()
+        if telescope not in self.available:
+            raise ValueError('the given telescope\'s field of view is unavailable at this time')
+
+        position = self.dim[telescope][0]
+        dimensions = self.dim[telescope][-1]
+        # dimensions is a list of lists [ [[ras], [decs]], '', ...]
+
+        # test that pos matches what the user entered
+        if center and position == 'absolute':
+            raise ValueError('the given telescope does not take center coordinates')
+        elif not center and position == 'relative':
+            raise ValueError('the given telescope requires center coordinates')
+
+        # draw jwst nircam's inner panels first
+        if telescope[-6:] == 'nircam':
+            self._gen_fov('nircam_short', center, rot, **kwargs)
+
+        for panel in dimensions:
+            ras = (panel[0] * u.deg).value
+            decs = (panel[1] * u.deg).value
+            ra_r, dec_r = self._rotate(ras, decs, rot)
```

Comment:
Rotations should only happen if ``position='relative'``?

Reply:
`rot` is 0 by default, and the plan with the checks on lines 285 and 287 was to prevent the case where an FOV with `position='absolute'` carries different rotation value besides 0. 

I see that this isn't currently the case, but if it was, then `self._rotate(ras, decs, rot)` where `rot` is 0 would just return `ras` and `decs`. That being the case (and for simplicity's sake), I figured running everything through `self._rotate()` was no harm no foul.

Does that sound OK, or would you rather not call the function if no rotation is necessary? I think it comes down to whether adding another `if/else` condition is more time efficient than running `self._rotate()`.

Reply:
I think we should just avoid calling the function at all if possible, for readability

Reply:
I missed this reply. So you're saying don't call `self.rotate()` unless `position == relative` and `rot != 0`?